initial commit to add ros2interface exporter

ali-momin12 opened the pull request:
Export a VSS model to a ROS 2 interface package: generates .msg files (per leaf or aggregated by parent branch) and optional .srv files for Get/Set operations. This exporter plugs into the vspec export CLI like other vss-tools exporters.

ali-momin12 committed:
initial commit to add ros2interface exporter

signed-off-by: Ali MOMIN <ali_momin@jp.honda>

ali-momin12 committed:
Documentation for ros2interface exporter

signed-off-by: Ali MOMIN <ali_momin@jp.honda>

ali-momin12 committed:
fixing pre-commit checks in md file of ros2 exporter

signed-off-by: Ali MOMIN <ali_momin@jp.honda>

shogo-kubota-esol commented on docs/vspec.md:
In the "Exporter Specific Documentation", please add a link to "ros2interface.md".
<img width="597" height="116" alt="image" src="https://github.com/user-attachments/assets/d3e0d276-6f1d-4f8c-b8fe-cd26902a1ebf" />

ali-momin12 replied:
fixed in the commit [5f955](https://github.com/ali-momin12/vss-tools/pull/2/commits/5f9555325c3f7c168925a5261ab21e86efa0160b)

shogo-kubota-esol replied:
I've confirmed the fix.

shogo-kubota-esol commented on src/vss_tools/exporters/ros2interface.py:
There is a difference compared to the contents of the following branches.
<https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/blob/feature/ros2-idl-intra-honda/src/vss_tools/exporters/ros2interface.py>

What causes this difference?

ali-momin12 replied:
no changes in the code, just a different formatter and some wording changes in the comments

shogo-kubota-esol replied:
understood.

shogo-kubota-esol commented on tests/vspec/test-ros2-interface.py:
There is a difference compared to the contents of the following branches.
<https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/blob/feature/ros2-idl-intra-honda/tests/vspec/test-ros2-interface.py>

What causes this difference?
Is it due to the fact that the formatter is different?

ali-momin12 replied:
yes, the formatter is different, its called pre-commit

shogo-kubota-esol replied:
understood.
I'll resolve this conversation.

shogo-kubota-esol commented on tests/vspec/test_generic.py:
The amendments to this file were not included in the original PR revisions.
<https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/blob/feature/ros2-idl-intra-honda/tests/vspec/test_generic.py>

Is this a fix made to pass the automated test?
If so, no problem.

ali-momin12 replied:
yes, these are the changes to help pass automated tests

shogo-kubota-esol replied:
understood.
I'll resolve this conversation.

shogo-kubota-esol commented:
In the following branch, "test_quantities.yaml" and "test_units.yaml" were also fixed.
<https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/pull/3/files#diff-7bbde7ad621ac78e09e7cbc23cd614e6befe376ede5022e8944c738d0303678b>
<https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/pull/3/files#diff-9ec77ba49a10e9fe0d096b5a81091089302c3f564f3df2be8055335aacc49509>

Are these fixes unnecessary?

shogo-kubota-esol commented:
expected.ros2interface folder is being created, what kind of folder is it?

ali-momin12 committed:
adding document link to vspec.md, adding some units and quantities in the test files to cope with the custom test file for ros2 exporter

signed-off-by: Ali MOMIN <ali_momin@jp.honda>

ali-momin12 commented:
> expected.ros2interface folder is being created, what kind of folder is it?

its a necessary folder which contains sample output files, crucial to pass automated tests

ali-momin12 commented:
> In the following branch, "test_quantities.yaml" and "test_units.yaml" were also fixed. https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/pull/3/files#diff-7bbde7ad621ac78e09e7cbc23cd614e6befe376ede5022e8944c738d0303678b https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/pull/3/files#diff-9ec77ba49a10e9fe0d096b5a81091089302c3f564f3df2be8055335aacc49509
> 
> Are these fixes unnecessary?

thanks for catching these missing quantities and units, i have fixed these in [5f955](https://github.com/ali-momin12/vss-tools/pull/2/commits/5f9555325c3f7c168925a5261ab21e86efa0160b)

shogo-kubota-esol commented:
> > In the following branch, "test_quantities.yaml" and "test_units.yaml" were also fixed. https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/pull/3/files#diff-7bbde7ad621ac78e09e7cbc23cd614e6befe376ede5022e8944c738d0303678b https://github.com/dpg-users-dev/COVESA-VSS-TOOLS/pull/3/files#diff-9ec77ba49a10e9fe0d096b5a81091089302c3f564f3df2be8055335aacc49509
> > Are these fixes unnecessary?
> 
> thanks for catching these missing quantities and units, i have fixed these in [5f955](https://github.com/ali-momin12/vss-tools/pull/2/commits/5f9555325c3f7c168925a5261ab21e86efa0160b)

I've confirmed the fix.

shogo-kubota-esol commented on docs/ros2interface.md:
````diff
@@ -0,0 +1,113 @@
+## Usage
+
+```bash
+vspec export ros2interface   --vspec spec/VehicleSignalSpecification.vspec   -I spec   --output ./out   --package-name vss_interfaces   --mode aggregate|leaf   --srv none|get|set|both   [--srv-use-msg | --no-srv-use-msg]   [--topics PATTERN ...]   [--exclude-topics PATTERN ...]   [--topics-file patterns.txt]   [--topics-case-insensitive]
````
This is fine on preview, but we should add "'''" at the end of the sentence.

<img width="690" height="189" alt="image" src="https://github.com/user-attachments/assets/81455117-3117-478e-84f1-3af37a04650a" />

ali-momin12 replied:
somehow with those extra  "'''" at the end, the pre-commits formatting check was failing again and again
thats why i delete that  "'''"

shogo-kubota-esol replied:
Are you putting a line break at the end of the file?
If you don't include a line break, the closing tag "```" may be automatically removed.

ali-momin12 replied:
fixed in the commit [a323dd4](https://github.com/ali-momin12/vss-tools/pull/2/commits/a323dd40c1ad4a394b9cf8f0d3270cf6ab71e26a)

shogo-kubota-esol replied:
I've confirmed the fix.

Ideally, the markdown file should always have a line break at the end.

shogo-kubota-esol reviewed (changes requested):
I checked again and found the difference.
Could you please check it out?

ali-momin12 committed:
fixing the missing ending quotations

signed-off-by: Ali MOMIN <ali_momin@jp.honda>

shogo-kubota-esol commented:
Could you please add the following to the PR description:

- [Standard Build Check / buildtest (pull_request)](https://github.com/ali-momin12/vss-tools/actions/runs/18156590164/job/51677757754?pr=2) fails because the required test coverage rate is more than 88% and the current test coverage rate for ros2interface is 55%
- Improving the coverage rate will be done in a PR to the main repository, and the results of the Build Check test will not be considered in this PR.

If the above explanation is added to the PR, I will approve the PR.

shogo-kubota-esol commented:
When merging this PR, please select "Squash and merge".
The reason for this is that, according to the comments below, the commit description requires a Signed-off-by, and the commits in the history now do not have a Signed-off-by mention.
<https://github.com/COVESA/vss-tools/issues/464#issuecomment-3317200081>
<img width="695" height="45" alt="image" src="https://github.com/user-attachments/assets/c40e6e8c-da8f-4ee0-a478-d232fb858e69" />

If you select "Squash and merge", you have to edit the commit name and the body of the commit, so please include your Signed-off-by mention in the body of the commit.
https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for

Example)  
https://github.com/ali-momin12/vss-tools/commit/614e5ada279c4abe13488e4aaabd080dc0ebd05d
<img width="561" height="194" alt="image" src="https://github.com/user-attachments/assets/d90267a8-37df-424f-8855-3be05478a3ad" />

Also, if you write `#464`, it will automatically be associated with an issue.
So could you include it in the commit name?
Example) #464

ali-momin12 committed:
adding expected output files for ros2exporter in test_comments, test_isinstance, test_min_max tests for better test coverage

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
checking a prospected test coverage issue

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding file comparison checks for ros2exporter

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding extra params for ros2 exporter to be checked in generic testing

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding optional params for ros2 exporter in generic test file

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding some more optional params for ros2 exporter in generic test file

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
fixing spelling mistake

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
changing parms to increase test coverage

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
changing parms to increase test coverage ratio

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
changing parms again

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding more params to test ros2 exporter

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding more params

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding more params

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding more test coverage by eliminating unnecessary checks

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
deleting some unnecessary try check blocks so that test coverage can improve

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
tweaking the generic test file to increase ros2 exporter coverage

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
rolling back last commit changes

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
restructring the generic test file to accomodate the complex test coverage scenarios of ros2 exporter

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
fixing structural issues in generic test file

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding fix for exporter run subcommand

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
fixing some runtime syntax errors

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
rolling back changes to original strucrture for generic test file, adding some simple commands to dually test ros2 exporter both in leaf and aggregate mode for better test coverage ratio

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
changing the exporter call method in generic test file

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
changing the test case for ros2 exporter

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
dual exporter test calling

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
rolling back to last commit, removing some optinal params from ros2 exporter

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
changing the aggregate call in ros2 exporter to also generate leaf level outputs, its just to check the better test coverage

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
changing the generic test file code for ros2 exporter to execute as many optional params as possible to gain better test coverage results

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
changing param for ros2 exporter in generic test call

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
stripping the sanitization of comment value code from ros2 interface, for better test coverage

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
adding repeatable topics selection in the ros2 exporter cli execution syntax to get better test coverage results

Signed-off-by: Ali Momin <ali_momin@jp.honda>

ali-momin12 committed:
More vss topics added in the ros2 exporter cli execution syntax to get better test coverage results

Signed-off-by: Ali Momin <ali_momin@jp.honda>
